packages/cli-hooks/src/protocols.spec.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-console */
+
+import { afterEach, beforeEach, describe, it } from 'mocha';
+import assert from 'assert';
+import sinon from 'sinon';
+
+import {
+  BaseProtocol,
+  MessageBoundaryProtocol,
+  getProtocolInterface,
+} from './protocols.js';
+
+describe('protocol implementations', () => {
+  describe('default protocol', () => {
+    it('stubs logging methods with a manifest flag', () => {
+      const protocol = BaseProtocol(['--manifest']);
+      assert.notEqual(protocol.log, console.log);
+      assert.notEqual(protocol.error, console.error);
+      assert.notEqual(protocol.warn, console.warn);
+      assert.equal(protocol.respond, console.log);
+    });
+
+    it('uses console log methods without a manifest', () => {
+      const protocol = BaseProtocol(['--flag']);
+      assert.equal(protocol.log, console.log);
+      assert.equal(protocol.error, console.log);
+      assert.equal(protocol.warn, console.log);
+      assert.equal(protocol.respond, console.log);
+    });
+  });
+
+  describe('message boundary protocol', () => {
+    /** @type {sinon.SinonStub} */
+    let consoleLogStub;
+
+    beforeEach(() => {
+      consoleLogStub = sinon.stub(console, 'log');
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('errors if no boundary is specified', () => {
+      assert.throws(() => {
+        MessageBoundaryProtocol([]);
```

Comment:
I'd add an assertion here on the error text, in case we introduce a different kind of bug in the future raising a different kind of error.